# Initialize database files

In [1]:
%cd twitteranalysis
import environment
import sqlite3
import os
from Loggers.Helpers import delete_files


g = environment.sqlite_file_connection_string_generator()
query = """CREATE TABLE `word_map_deux` (
`tweet_id` int(20) DEFAULT NULL,
  `user_id` int(20) DEFAULT NULL,
  `word` varchar(200) DEFAULT NULL,
  `sentence_index` int(11) DEFAULT NULL,
  `word_index` int(11) DEFAULT NULL)"""

# Make master db and tables
import sqlite3

# make individual tables
def initialize_working_tables():
    for i in range(0, environment.MAX_DB_FILES +1):
        try:
            f = '%s/wordmapping%s.db' % (environment.DB_FOLDER, i)
            conn = sqlite3.connect(f)
            conn.execute(query)
            conn.commit()
            conn.close()
        except:
            pass

def initialize_master_db():
    try:
        conn = sqlite3.connect(environment.MASTER_DB)
        conn.execute(query)
        conn.commit()
        conn.close()
    except:
        pass

    
def delete_working_files():
    delete_files(environment.DB_FOLDER)

(bookmark:twitteranalysis) -> /Users/adam/Dropbox/PainNarrativesLab/TwitterDataAnalysis
/Users/adam/Dropbox/PainNarrativesLab/TwitterDataAnalysis


In [2]:
delete_working_files()

removing wordmapping3.db
removing wordmapping7.db
removing wordmapping6.db
removing wordmapping2.db
removing wordmapping9.db
removing wordmapping8.db
removing wordmapping5.db
removing wordmapping10.db
removing wordmapping1.db
removing wordmapping0.db
removing wordmapping4.db


In [3]:
initialize_master_db()

In [4]:
initialize_working_tables()

In [ ]:
%


# Run the user parser

In [5]:
%cd twitteranalysis
%run -i Executables/process_user_descriptions_into_words2.py

(bookmark:twitteranalysis) -> /Users/adam/Dropbox/PainNarrativesLab/TwitterDataAnalysis
/Users/adam/Dropbox/PainNarrativesLab/TwitterDataAnalysis
ready to run


RuntimeError: This event loop is already running

connection ready
about to process
4352 users processed (not nec done)
done processing


In [ ]:
class AsyncBatcher(object):
    __slots__ = ['batch', 'batch_size', 'save', 'flush']
    def __init__(self, batch_size):
        self.batch_size = batch_size
        self.batch = []
    
    

In [ ]:
%cd twitteranalysis
from Executables.process_user_descriptions_into_words2 import main
main()


In [ ]:
%cd twitteranalysis
from Executables.process_user_descriptions_into_words2 import main
main()


In [6]:
%cd twitteranalysis

from Servers.ClientSide import Client
c = Client()

# Each of the listening request handlers needs its queue flushed
c.send_flush_command()

j = c.send_shutdown_command()

(bookmark:twitteranalysis) -> /Users/adam/Dropbox/PainNarrativesLab/TwitterDataAnalysis
/Users/adam/Dropbox/PainNarrativesLab/TwitterDataAnalysis


Future exception was never retrieved
future: <Future finished exception=HTTP 500: Internal Server Error>
tornado.httpclient.HTTPError: HTTP 500: Internal Server Error


In [ ]:
j

# Merge the various sqlite files

In [ ]:

%cd twitteranalysis
import environment
import sqlite3

mainDB = environment.MASTER_DB
columnNames = "word_index, sentence_index, word, user_id, tweet_id"
    
otherDbNames = [ 'wordmapping%s.db' % (i) 
            for i in range( 0, environment.MAX_DB_FILES +1)]

otherDBs = [ '%s/%s' % (environment.DB_FOLDER, name) 
            for name in otherDbNames]
    
for db in otherDbNames:
    conn = sqlite3.connect( mainDB )  # Connect to the main database
    curs = conn.cursor()  # Connect a cursor
    
    dbPath = '%s/%s' % (environment.DB_FOLDER, db)
    curs.execute( "ATTACH DATABASE ? as ? ;", (dbPath, 'db') )

    dbNameTableName = "db.word_map_deux"
    tableName = 'word_map_deux'
    curs.execute( "INSERT INTO %s (%s) SELECT %s FROM %s;" % (tableName, columnNames, columnNames, dbNameTableName) )
    conn.commit()
    curs.close()
    conn.close()


# Check data

In [ ]:
%cd twitteranalysis
import environment
import sqlite3

In [ ]:
conn = sqlite3.connect(environment.MASTER_DB)
r1 = conn.execute("select count( distinct user_id) from word_map_deux")
print("%s Unique user ids" % r1.fetchone())
r2 = conn.execute("select count( word) from word_map_deux")
print("%s rows in master.db" % r2.fetchone())
conn.close()

Before changed to class method w separate call to flush
    3591 Unique user ids
    47952 rows in master.db

After
    4014 Unique user ids
    67940 rows in master.db
    
NB: 
    numberUsers processed = 4352

    users w empty descriptions = 332
    
    users non-english = 129

In [ ]:
4014+332

In [ ]:
conn = sqlite3.connect(environment.MASTER_DB)
r1 = conn.execute("select user_id, count(word) from word_map_deux group by user_id ")
print( r1.fetchone())
conn.close()

In [ ]:
conn = sqlite3.connect(environment.MASTER_DB)
curs = conn.cursor()  # Connect a cursor
    
r1 = curs.execute("select * from word_map_deux where user_id = 1460971")
# for i in range(0, 100):
#     print( r1.fetchone())

curs.close()

In [ ]:
"""The above tells me that the problem is the whole store getting sent to the server"""

# Explore data

In [ ]:
def master_row_generator():
    conn = sqlite3.connect(environment.MASTER_DB)
    r = conn.execute("select * from word_map_deux")
    while True:
        yield r.fetchone()

def db_row_generator(filepath):
    conn = sqlite3.connect(filepath)
    r = conn.execute("select * from word_map_deux")
    while True:
        yield r.fetchone()


In [ ]:
gen = master_row_generator()

In [ ]:
for i in range(0, 220):
    print(next(gen))

In [ ]:
f = '/Users/adam/Desktop/TwitterDataAnalysisLogs/dbs/wordmapping1.db'
dgen = db_row_generator(f)
for i in range(0, 220):
    print(next(dgen))

In [ ]:
f = '/Users/adam/Desktop/TwitterDataAnalysisLogs/dbs/wordmapping1.db'
conn = sqlite3.connect(f)
r = conn.execute("select * from word_map_deux")
for i in range(0, 5):
    print(r.fetchone())


In [ ]:
f = '/Users/adam/Desktop/TwitterDataAnalysisLogs/dbs/wordmapping9.db'
conn = sqlite3.connect(f)
r = conn.execute("select * from word_map_deux")
for i in range(0, 5):
    print(r.fetchone())


In [ ]:

f = '/Users/adam/Desktop/TwitterDataAnalysisLogs/master.db'
conn = sqlite3.connect(f)
r = conn.execute("select count(word) from word_map_deux")
print(r.fetchone())
conn.close()

# Timing

In [ ]:
%cd twitteranalysis

from functools import wraps
import time

from Loggers.CsvLoggers import log_query, log_query_timestamp
import environment


def time_and_log(fn):
    """
    Decorator to time operation of method
    From High Performance Python, p.27
    """
    @wraps(fn)
    def measure_time(*args, **kwargs):
        t1 = time.time()
        result = fn(*args, **kwargs)
        t2 = time.time()
        t2 = time.time()
        elapsed = t2 - t1
        log_query( elapsed )
        log_query_timestamp()
#         print(("@timefn:%s took %s seconds" % (fn.__name__, elapsed)))
        return result
    return measure_time


In [ ]:

@time_and_log
def test():
    time.sleep(2)
    print('done')
    

In [ ]:
test()